### PR TITLE
[Feature] added a filter for games

### PR DESCRIPTION
**Pull request by LordOfRhun**
### Description
this add the feature to filter by game names.

### Specifications
The comboBox is feeded by the _Wabbajack.Common.Game_ enum so any newly added games should be in the filter automatically. the filter works by simply comparing the metadata from the game and the selected game from the filter.

### Preview
https://i.imgur.com/lVSK7DB.gifv

my discord tag is _Khamûl#0128_ for communication

### CODE REVIEW

**Comment by Noggog** [comment]
Looking solid!  Initially I was going to comment on maybe keeping the Enum around longer instead of having a list of strings, but given that we need "All" to be an option on the list, that complicates things.  I think the string solution is a good setup for now with the current featureset.

It's looking good to me

File: Wabbajack/Views/ModListGalleryView.xaml [hunk 1]
```diff
@@ -139,4 +151,4 @@
             <iconPacks:PackIconMaterial Foreground="{Binding Foreground, RelativeSource={RelativeSource AncestorType={x:Type Button}}}" Kind="ArrowLeft" />
         </Button>
     </Grid>
-</rxui:ReactiveUserControl>
+</rxui:ReactiveUserControl>
```
Comment (Noggog) [inline Wabbajack/Views/ModListGalleryView.xaml:154]: Git gets unhappy without an extra new line at the end.  Not sure why.
Smallest of nothing issues.. just happened to notice it was intentionally removed while scrolling through

**Comment by Noggog** [comment]
Maybe add a `const string` member to store "All".  Just makes it more robust if we want to change the term later

**Comment by LordOfRhun** [comment]
I made some changes from your suggestions, and the games names now comes from the enum description attribute as suggested by @erri120 on discord. It looks much better

**Pull request merged**